### PR TITLE
Add network topologies

### PR DESCRIPTION
**Pull request by jverbraeken**

### CODE REVIEW

File: results.tex [hunk 1]
```diff
@@ -88,7 +88,7 @@ \subsubsection{Partitioning and distribution algorithms}
 
 
 \paragraph{Bridging computation and communication}
-Let's look at several ways to efficiently communicate between the nodes taking the interleaving of parallel program computations and inter-worker communication into account.
+Let's look at several ways to efficiently communicate between the nodes taking the interleaving of parallel program computations and inter-worker communication into account. These techniques trade-off communication overhead (at the top of the list) with fresher updates (at the bottom of the list).
```
Comment (Kloppie5) [inline results.tex]: trade-off is a noun, if used as a verb the hyphen is removed

File: results.tex [hunk 2]
```diff
@@ -106,17 +106,25 @@ \subsubsection{Partitioning and distribution algorithms}
...
+	An advantage is that all model parameters are in a global shared memory which makes it easy to inspect the model. A disadvantage is that the parameter servers are a bottleneck because they're handling all communication. To partly accommodate this issue, the techniques mentioned under "Bridging computation and communication" are used.
+	\item \textbf{Decentralized storage} In contrast to centralized storage, in decentralized storage every worker maintains its own local view of the parameters and the workers communicate directly to each other. An example implementation is a peer-to-peer network where every machine broadcasts everything to all other machines. To reduce the amount of communication between all workers, Sufficient Factor Broadcasting (SFB)\cite{Li13}) was proposed. It Decomposes the parameter matrix \textbf{W} into the outer products of the two vectors \textbf{u} and \textbf{v}: \textbf{ΔW = uv\textsuperscript{T}}. The vectors u and v are \textit{sufficient factors}, meaning that they are sufficient to reconstruct the update matrix \textbf{ΔW}. SFB only broadcasts the sufficient factors and lets the workers reconstruct the updates.\\
```
Comment (Kloppie5) [inline results.tex]: Do we need to explain why decomposing a matrix into vectors would reduce the amount of data needed to broadcast and the trade-off with reconstructing the matrix?

**Comment by jverbraeken** [comment]
Implemented the suggestions of @Kloppie5

**Review by jkatzy — APPROVED** [review]
lgtm

**Pull request merged**